packages/vaex-core/vaex/__init__.py

```diff
@@ -474,13 +475,99 @@ def from_json(path_or_buffer, orient=None, precise_float=False, lines=False, cop
+        >>>     df.export_hdf5(f'taxi_{i:02}.hdf5')
+
+    :param bool or str convert: convert files to an hdf5 file for optimization, can also be a path. The CSV
+        file will be read in chunks: either using the provided chunk_size argument, or a default size. Each chunk will
+        be saved as a separate hdf5 file, then all of them will be combined into one hdf5 file. So for a big CSV file
+        you will need at least double of extra space on the disk. Default chunk_size for converting is 5 million rows,
+        which corresponds to around 1Gb memory on an example of NYC Taxi dataset.
+    :param kwargs: extra keyword arguments, currently passed to Pandas read_csv function, but the implementation might
+        change in future versions.
+    :returns: DataFrame
+    """
+    if not convert:
+        return _from_csv_read(filename_or_buffer=filename_or_buffer, copy_index=copy_index,
+                              chunk_size=chunk_size, **kwargs)
+    else:
+        if chunk_size is None:
+            # make it memory efficient by default
+            chunk_size = 5_000_000
+        return _from_csv_convert_and_read(filename_or_buffer=filename_or_buffer, copy_index=copy_index,
+                                          maybe_convert_path=convert, chunk_size=chunk_size, **kwargs)
+
+
+def _from_csv_read(filename_or_buffer, copy_index, chunk_size, **kwargs):
+    import pandas as pd
+    if not chunk_size:
+        full_df = pd.read_csv(filename_or_buffer, **kwargs)
+        return from_pandas(full_df, copy_index=copy_index)
+    else:
+        def iterator():
+            chunk_iterator = pd.read_csv(filename_or_buffer, chunksize=chunk_size, **kwargs)
+            for chunk_df in chunk_iterator:
+                yield from_pandas(chunk_df, copy_index=copy_index)
+        return iterator()
+
+
+def _from_csv_convert_and_read(filename_or_buffer, copy_index, maybe_convert_path, chunk_size, **kwargs):
+    # figure out the CSV file path
+    if isinstance(filename_or_buffer, six.string_types):
+        csv_path = filename_or_buffer
+    elif isinstance(maybe_convert_path, six.string_types):
+        csv_path = re.sub(r'\.hdf5$', '', str(maybe_convert_path), flags=re.IGNORECASE)
+    else:
+        raise ValueError('Cannot derive filename to use for converted HDF5 file, '
+                         'please specify it using convert="my.csv.hdf5"')
+
+    # reuse a previously converted HDF5 file
+    import vaex.file
+    combined_hdf5 = _convert_name(csv_path)
+    if os.path.exists(combined_hdf5):
+        return vaex.file.open(combined_hdf5)
+
+    # convert CSV chunks to separate HDF5 files
     import pandas as pd
-    return from_pandas(pd.read_csv(filename_or_buffer, **kwargs), copy_index=copy_index)
+    converted_paths = []
+    csv_reader = pd.read_csv(filename_or_buffer, chunksize=chunk_size, **kwargs)
+    for i, df in enumerate(csv_reader):
+        ds = from_pandas(df, copy_index=copy_index)
```

Comment:
before vaex had a dataframe API, it's classname was a dataset, and we'd use `ds` a lot in the codebase. We're moving that to `df` everywhere when we refactor/change. I think it's a good chance to do this here as well, and give pandas dataframes a `_pandas` suffix.
```suggestion
        df = from_pandas(df, copy_index=copy_index)
```

Reply:
Ok, will do.